partitionmanager/bootstrap.py

```diff
@@ -130,15 +231,22 @@ def calculate_sql_alters_from_state_info(conf, in_fp):
             log.info(f"Skipping {table_name} as it is not in the current config")
             continue
 
-        problem = pm_tap.get_table_compatibility_problems(conf.dbcmd, table)
-        if problem:
-            raise Exception(problem)
+        map_data = None
+
+        if not conf.assume_partitioned_on:
+            problem = pm_tap.get_table_compatibility_problems(conf.dbcmd, table)
+            if problem:
+                raise Exception(problem)
+            map_data = pm_tap.get_partition_map(conf.dbcmd, table)
+        else:
+            map_data = _override_config_to_map_data(conf)
```

Comment:
This logic occurs here and in `write_state_info`. Perhaps it could be refactored into a method on `conf`? That might also make it easier to introduce a test to ensure that the value returned from `_override_config_to_map_data` is a valid partition map.

Reply:
Done